demo/api/src/products/entities/product.entity.ts

```diff
@@ -42,6 +44,10 @@ export class Product extends BaseEntity<Product, "id"> implements DocumentInterf
     @Field()
     inStock: boolean = true;
 
+    @Property({ customType: new RootBlockType(DamImageBlock) })
+    @Field(() => GraphQLJSONObject, { nullable: true }) //TODO should not be nullable
+    image: BlockDataInterface;
```

Comment:
Should we add the `@RootBlock` and `@RootBlockEntity` decorator to the entity? Also, you could use the RootBlockData scalar

Reply:
yes